consensus/src/main/java/org/apache/iotdb/consensus/IStateMachine.java

```diff
@@ -69,6 +69,18 @@ default boolean isReadOnly() {
    */
   boolean takeSnapshot(File snapshotDir);
 
+  /**
+   * Take a snapshot of current statemachine. Snapshot.log will be stored under snapshotDir, The
+   * data of the snapshot will be stored under `data folder/snapshot/snapshotId`.
+   *
+   * @param snapshotDir required storage dir
+   * @param snapshotId the id of the snapshot
+   * @return true if snapshot is successfully taken
+   */
+  default boolean takeSnapshot(File snapshotDir, String snapshotId) {
```

Comment:
remove this default implementation and implement this method in all sub-classes

Reply:
Other implementation classes do not need to implement the new method, changing the default implementation to call the original method